### ISSUE TITLE
redshift dialect: parse error while comparing column values in where clause

### ISSUE THREAD

**Issue by 6e2f61** [sqlfluff/sqlfluff#3512]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Querying redshift system tables produces parsing error in where clause while comparing values.

### Expected Behaviour

Add support for the redshift system table select query syntax.

### Observed Behaviour

Part of query not currently supported, results in parse errors. Schema somehow is omitted by sqlfluff
```bash
== [app/sys_tbl_redshift_part1.sql] FAIL                                                                                                                                                             
L:   7 | P:   5 | L027 | Unqualified reference 'pg_namespace' found in select with more than one referenced table/view.
L:   7 | P:  17 |  PRS | Line 7, Position 17: Found unparsable section: '.oid = pg_constraint.connamespace'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Create a test file containing as below and try to lint/parse with sqlfluff.
```sql
SELECT
    pg_namespace.nspname AS constraint_schema
    , pg_constraint.conname AS constraint_name
FROM
    pg_namespace, pg_constraint
WHERE
    pg_namespace.oid = pg_constraint.connamespace
```

### Dialect

redshift

### Version

`sqlfluff, version 1.0.0`
`Python 3.8.10`

### Configuration

```bash
# https://docs.sqlfluff.com/en/stable/configuration.html#config

[sqlfluff]
dialect = redshift
templater = jinja
rules = all
exclude_rules = L034
# Controls SQLFluff output, see max_line_length for SQL output
output_line_length = 118
ignore_templated_areas = true
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = true

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 118
indent_unit = space
comma_style = trailing

# https://docs.sqlfluff.com/en/stable/rules.html#ruleref

[sqlfluff:rules:L003]
hanging_indents = true

[sqlfluff:rules:L007]
operator_new_lines = after

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper

[sqlfluff:rules:L011]
# of tables
aliasing = explicit

[sqlfluff:rules:L012]
# of columns
aliasing = explicit

[sqlfluff:rules:L013]
allow_scalar = false

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
ignore_comment_clauses = true
ignore_comment_lines = true
indent_unit = space
max_line_length = 160
tab_space_size = 4

[sqlfluff:rules:L019]
comma_style = leading

[sqlfluff:rules:L022]
comma_style = leading

[sqlfluff:rules:L029]
unquoted_identifiers_policy = all
ignore_words = character_maximum_length, column_name, columns, constraint_name, constraint_schema, data_type, role, status, table_name, type

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions
force_enable = true

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L051]
fully_qualify_join_types = both

[sqlfluff:rules:L052]
multiline_newline = false
require_final_semicolon = false

[sqlfluff:rules:L054]
group_by_and_order_by_style = explicit

[sqlfluff:rules:L057]
additional_allowed_characters = "{}"
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = false

[sqlfluff:rules:L059]
prefer_quoted_identifiers = false

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by 6e2f61** [comment]
@MrocznyKapelusz

**Comment by tomasfarias** [comment]
Hey all, I ran into this issue while linting some new views we are creating to audit our Redshift database.

The root of the issue is that `oid` is a Redshift keyword (as defined [here](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_redshift_keywords.py#L102)). Attempting to use it as a column reference causes the unparsable error, even though using it as a **qualified** reference is completely valid.

I dug through the issues and noticed the same problem has already been solved for the Postgres dialect (see [issue](https://github.com/sqlfluff/sqlfluff/issues/3412) and [PR](https://github.com/sqlfluff/sqlfluff/pull/3450). Given that `oid` is a common column identifier in several system tables/views and that [AWS examples](https://docs.aws.amazon.com/redshift/latest/dg/c_join_PG_examples.html) frequently use it as a qualified reference, I went ahead and opened a PR that implements the same solution as was done for the Postgres dialect to support it.
